From 8ea19fc840a7156b3c324d67d28282dd2f37f55a Mon Sep 17 00:00:00 2001
From: Donna Tom <donna@heliumfoot.com>
Date: Tue, 7 Oct 2025 19:06:54 -0700
Subject: [PATCH] Support 16Kb page sizes for Android

---
 stdlib/cmake/modules/AddSwiftStdlib.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/stdlib/cmake/modules/AddSwiftStdlib.cmake b/stdlib/cmake/modules/AddSwiftStdlib.cmake
index 77dff0e1a21..9436dffd41a 100644
--- a/stdlib/cmake/modules/AddSwiftStdlib.cmake
+++ b/stdlib/cmake/modules/AddSwiftStdlib.cmake
@@ -2452,6 +2452,8 @@ function(add_swift_target_library name)
         list(APPEND swiftlib_link_flags_all "-Wl,-soname,lib${name}.so")
         # Add the build-id flag to ensure a unique identifier is included in the output binary.
         list(APPEND swiftlib_link_flags_all "-Wl,--build-id")
+         # Ensure compatibility with Android 15+ devices using 16KB memory pages.
+         list(APPEND swiftlib_link_flags_all "-Wl,-z,max-page-size=16384")
       endif()
 
       if (SWIFTLIB_BACK_DEPLOYMENT_LIBRARY)
-- 
2.39.5 (Apple Git-154)

